src/Kestrel.Core/Internal/Http2/HPack/DynamicTable.cs

```diff
@@ -7,23 +7,25 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.HPack
 {
     public class DynamicTable
     {
-        private readonly HeaderField[] _buffer;
-        private int _maxSize = 4096;
+        private HeaderField[] _buffer;
+        private int _maxSize;
         private int _size;
         private int _count;
         private int _insertIndex;
         private int _removeIndex;
 
         public DynamicTable(int maxSize)
         {
-            _buffer = new HeaderField[maxSize];
+            _buffer = new HeaderField[maxSize / 32];
```

Comment:
The header field names and values need to be at least one octet too, right? Could you divide by 34?

Reply:
The value may be empty.